torch/_dynamo/side_effects.py

```diff
@@ -435,7 +433,7 @@ def codegen_save_tempvars(self, cg: PyCodegen):
                 )
                 cg.extend_output(create_call_function(0, False))
                 cg.add_cache(var)
-                var.mutable_local.source = LocalSource(cg.tempvars[var])  # type: ignore[attr-defined]
+                var.source = LocalSource(cg.tempvars[var])  # type: ignore[attr-defined]
```

Comment:
Trying to understand. Overriding source is OK here because we have already generated guards?

Reply:
Ah so here since we are dealing with temporary variables for which Dynamo supports mutation, e.g.,
```python
def f(x, y):
  tmp_obj = MyObject()
  tmp_obj.foo = x + y
  ...
```
so we don't need guards for `tmp_obj`.

With that I see `Source` being used for 2 purposes in Dynamo:
1. Help generate guards (this is probably a must-have)
2. Help with codegen (this seems to be an impl decision; we could tweak things by, e.g., storing this information as a `Dict[Source, str] source_to_tempvar_name`...)

The use case here is (2). I'll send a subsequent PR to beef up all these documentations, and maybe refactor the impl if that can makes things more clear.